src/main/java/it/gov/pagopa/paymentupdater/deserialize/CustomDateTimeDeserializer.java

```diff
@@ -0,0 +1,50 @@
+package it.gov.pagopa.paymentupdater.deserialize;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class CustomDateTimeDeserializer extends JsonDeserializer<LocalDateTime> {
+  /**
+   * The main assumption for this deserializer is that Payment Updater manages
+   * only UTC date times.
+   * It transforms millisecond based timestamps to second based timestamp by
+   * removing all digits above 10th position and it simplifies ISO strings by
+   * removing the timezone since it is UTC based.
+   * @param jsonParser
+   * @param deserializationContext
+   * @return
+   * @throws IOException when a unmanaged format is present
+   */
+  @Override
+  public LocalDateTime deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+    String modelDateTime = jsonParser.getValueAsString();
+    LocalDateTime parsedTimeTime;
+    try {
+      // we parse the local date time from the milliseconds by taking the seconds
+      // and considering UTC timezone
+      parsedTimeTime = LocalDateTime.ofEpochSecond(Long.parseLong(modelDateTime.substring(0,10)), 0, ZoneOffset.UTC);
+    } catch (NumberFormatException nfex) {
+      try {
+        // we parse the local date time removing the timezone
+        parsedTimeTime = LocalDateTime.parse(modelDateTime.substring(0, 23));
+      } catch (Exception ex) {
+        log.error(ex.getMessage());
+        log.error("Found a not managed format for a date: " + modelDateTime);
+        throw ex;
```

Comment:
@michaeldisaro LocalDateTime could be fire a DateTimeParseException didn't declared in throws method sign and there is a general exception catch without specific Exception encapsulation. 
Could be cause a fired runtime exception not managed?
We could be throw a new IOException(ex) or change the method sign accordley with LocalDateTime.parse sign?

Reply:
Yes, it is possible and it is what I wanted. 
I mean, we are trying to fix the parsing of some dates that have a wrong format. 
So I thought, what if there are other formats that are going to break everything? 
Well, we propagate the exception but we add another error log to identify which format broke everything again.
```
log.error(ex.getMessage());
log.error("Found a not managed format for a date: " + modelDateTime);
```
This way I think we will be identifying and fixing it faster.